### ISSUE TITLE
v-repeat in tables with two rows per repeat

### ISSUE THREAD

**Issue by swift1** [vuejs/Discussion#295]
Hi guys! I have a bit of a problem with tables. Consider the following code:

````html
<table>
	<tbody>
		<tr v-repeat="row: rowFromServer">
			<td>{{ row.col1 }}</td>
			<td>{{ row.col2 }}</td>
			<td>{{ row.col3 }}</td>
		</tr>
	</tbody>
</table>
````

This works fine. However, after each row in the main table, I'd like to have a child-table on a new row, but on the same v-repeat. So I figured the only way to achieve this would be to put a parent element before the <tr> tags, like this:

````html
<table>
	<tbody>
		<div v-repeat="row: rowFromServer">
			<tr>
				<td>{{ row.col1 }}</td>
				<td>{{ row.col2 }}</td>
				<td>{{ row.col3 }}</td>
			</tr>
			<tr>
				<td colspan="3">{{row.childArray.someProperty}}</td>
			</tr>
		</div>
	</tbody>
</table>
````

However, this approach does not work. Any tips?

**Comment by simplesmiler** [comment]
According to the [W3C](http://www.w3.org/TR/html-markup/tr.html#tr-context), `tr` is allowed to be inside of `table`, `thead`, `tfoot` or `tbody`. Neither `div` nor `template` are valid parents to `tr`.

You could try to `v-repeat` on `tbody`. As far as I know, it is allowed for `table` to contain multiple `tbody`.

**Comment by swift1** [comment]
Yeah I know, that's why I was hoping for some repeater that was invisible to the DOM, but that could still contain a repeater. :) But you're right, I guess putting it in ````tbody```` is the only way (and then use a function or a filter instead of ````<td>{{item}}````).

**Comment by simplesmiler** [comment]
Usually `template` serves as an invisible parent, but it is not applicable in this case.

I think there is also another way to do this using components.
You could write a custom filter that outputs each item multiple times.
And then you can do something like this:
```html
<tr v-repeat="row: rows | multiple" v-component="{{ toRowComponent($index) }}"></tr>
```

What was your solution in the end?

**Comment by swift1** [comment]
Ye I was thinking about custom components too, but in the end, I realized that I was making the problem more complex than it actually was. At first I thought that the sub-sub element couldn't access the repeater's value without having a parent repeater, but when I realized that that was wrong, it was easy. The final solution was something like this:

````html
<table>
	<tbody v-repeat="row: array">
		<tr>
			<td>{{row.col1}}</td>
			<td>{{row.col2}}</td>
			<td><a href="{{row.link}}">{{row.col3}}</a></td>
		</tr>
		<tr v-repeat="elem: row.subArray">
			<td>{{elem}}</td>
		</tr>
	</tbody>
</table>
````

**Comment by Nirazul** [comment]
Why don't you change your data model instead of trying to bend the html syntax?

```
computed: {
    myRows: function() {
        var result = [];
        var intermediary = this.$data.rowFromServer.forEach(function(row) {
            result.push(row);
            result.push(row.childArray);
        });

        return result;
    }
}
```

and then in your template:
```
<table>
    <tbody>
        <div>
            <tr v-repeat="row: myRows">
                <td v-if="$index % 2 === 0"> {{ row.col1 }} </td>
                <td v-if="$index % 2 === 0"> {{ row.col1 }} </td>
                <td v-if="$index % 2 === 0"> {{ row.col1 }} </td>

                <td v-if="$index % 2 === 1" colspan="3">{{ row.someProperty }}</td>
            </tr>
            <tr>
                
            </tr>
        </div>
    </tbody>
</table>
```

It won't win the annual competition for most beautiful code, but it should do the trick.

**Comment by swift1** [comment]
Ye but I think the solution I ended up with is a better candidate for the annual code competition, so I'll stick with that one. :smile:
